### DISCUSSION TITLE
Example of sub commands with Args?

### DISCUSSION THREAD

**Discussion by blaisep** [pypyr/pypyr#287] [Q&A] [answered]
I couldn't  find an example to match the following scenario:
Some CLIs (notably `git`, `podman`, `docker`) have sub-commands and those sub-commands have arguments. For example:

```bash
 podman run --name api-server --detach --tty \
  --volume ~/.ara/server:/opt/ara:z -p 8000:8000 \
  docker.io/recordsansible/ara-api:latest
```

I tried to break this up several ways:
```bash
  - name: pypyr.steps.cmd
    comment: run the ara API server in a podman container
    in:
      cmd:
        'podman run \
        --name api-server \
        --detach \
        --tty \
        --volume ~/.ara/server:/opt/ara \
        -p 8000:8000 \
        docker.io/recordsansible/ara-api:latest'
```
as well as trying a multi line quote using `cmd: |-` which was horrible.
And even various permutations of:
```bash
  - name: pypyr.steps.cmd
    comment: run the ara API server in a podman container
    in:
      cmd:
        - podman
        - run:
          - --name api-server 
          - --detach
          - --tty
          - --volume ~/.ara/server:/opt/ara 
          - -p 8000:8000 
          - docker.io/recordsansible/ara-api:latest
       ```

**Comment by yaythomas** [comment, marked as answer]
Hi @blaisep, you were very almost there! Instead of `|-` (keep newlines, no newline at end) you should just be using `>` (folded, replace newlines w spaces).

```
  - name: pypyr.steps.cmd
    in:
      cmd: >
        echo
        one
        two
        three
```

This will pass `echo one two three` to the subprocess.

So this way you can break up your cmd over multiple lines in the yaml however you want, and then under the hood they will get folded back up into a single line when passed to the subprocess.


FYI, your second example would deffo not work, but it might be instructive to explain why. . .
when you go
```yaml
- name: pypyr.steps.cmd
   in:
     cmd:
       - cmd1 --arg
       - cmd2 --blah
       - cmd3 --arb --arg
```

This will run cmd1, and then cmd2, and then cmd3. In other words, these are separate commands that run separately - each of these spawn their own subprocess, if you want to get technical about it. It's not "one cmd" spread over multiple lines.

**Comment by blaisep** [comment]
Oh wow, TIL about  folding with `>`  !!

Thank you.

**Comment by yaythomas** [comment]
Thanks for the question @blaisep, this is a good one - I have added a section to the help docs explaining the same so hopefully future pypyristas with the same question can have an immediate answer! https://pypyr.io/docs/steps/cmd/#split-long-command-over-multiple-lines
